source-mysql/discovery.go

```diff
@@ -492,13 +492,22 @@ func parseEnumValues(details string) []string {
 	// and take submatch #1 which is the body of each string.
 	var opts []string
 	for _, match := range enumValuesRegexp.FindAllStringSubmatch(details, -1) {
-		var opt = match[1]
+		opts = append(opts, decodeMySQLString(match[1]))
+	}
+	return opts
+}
+
+// decodeStringMySQL decodes a MySQL-format single-quoted string (including
+// possible backslash escapes) and returns it in unquoted, unescaped form.
+func decodeMySQLString(qstr string) string {
+	if strings.HasPrefix(qstr, "'") && strings.HasSuffix(qstr, "'") {
```

Comment:
probably dumb question: is it possible to have inputs like `'this''`? Where the decoded version should be `this'` but here we'd return `this`

Reply:
That specific input shouldn't be possible. The strings we're "decoding" here are actually fully decode within the Vitess tokenizer and then _re-encoded_ using their `encodeSQLString()` operation which will only ever produce correctly formed outputs (in this case the extra single-quote would be backslash-escaped so it'd be `'this\''`).

Even if that input did occur, we only strip off a single instance of the leading and trailing single-quote before decoding any escapes within the string, so it would product the correct result (inasmuch as a malformed input has a correct result at all).